app/controllers/content_controller.rb

```diff
@@ -122,7 +122,9 @@ def new
     ).uniq
 
     # todo this is a good spot to audit to disable and see if create permissions are ok also
-    unless (current_user || User.new).can_create?(content_type_from_controller(self.class))
+    unless (current_user || User.new).can_create?(@content.class) \
+      || PermissionService.user_has_active_promotion_for_this_content_type(user: current_user, content_type: @content.class.name)
```

Comment:
This is a real pain to have to add these checks instead of baking them into the `can_create?` authorizers, but unfortunately `authority` doesn't retain context of what class is being checked for permissions, so we can't query promotions on that class within the authorizer -- and `authority` is unmaintained, so I doubt that'll change. We'll probably want to do a big permissions overhaul soon, which should clean all this up.